internal/services/sockets/handle_socket.go

```diff
@@ -0,0 +1,66 @@
+package sockets
+
+import (
+	"log/slog"
+	"net/http"
+	"sync"
+
+	"github.com/gorilla/websocket"
+)
+
+// Globals
+var clients = make(map[*websocket.Conn]bool)
+var streamChan = make(chan []byte, 100)
+var lock sync.Mutex
+var upgrader = websocket.Upgrader{
+	CheckOrigin: func(r *http.Request) bool {
+		return true //Allows all origins for now
+	},
+}
+
+// HandleSocket will accept websocket connections on '/ws'
+func HandleSocket(mux *http.ServeMux) {
+	//Streams data to all connected clients
+	//TODO: Make it possible to only broadcast to specific groups of users.
+	go func() {
+		for data := range streamChan {
+			for conn := range clients {
+				err := conn.WriteMessage(websocket.TextMessage, data)
+				if err != nil {
+					slog.Error("Error sending to client", "reason", err)
+					lock.Lock()
+					delete(clients, conn)
+					lock.Unlock()
+				}
+			}
+		}
+	}()
+	mux.HandleFunc("/ws", func(w http.ResponseWriter, r *http.Request) {
+		var wg sync.WaitGroup
```

Comment:
Go's handler patterns can include a method at the beginning. this is relatively new feature. websocket requests are really just GET requests that get upgraded to long lived connections using some special headers.

It would be good to be able to identify connect clients. we will need to authorize these requests later. for now, lets just trust connects are who they say they are for the sake of being able to develop forward.

```suggestion
	mux.HandleFunc("GET /ws/{clientID}", func(w http.ResponseWriter, r *http.Request) {
		clientID := r.PathValue("clientID")
		var wg sync.WaitGroup
```

https://pkg.go.dev/net/http#Request.PathValue